### PR TITLE
Cleanup storehaus-cache

### PR DESCRIPTION
**Pull request by BenFradet**
This supersedes #309 and cleans up storehaus-cache.

### CODE REVIEW

**Comment by BenFradet** [comment]
@johnynek The mima checks fail, I'm still investigating how to deal with them.

**Comment by BenFradet** [comment]
- method nextSide()com.twitter.storehaus.cache.SideC# in object com.twitter.storehaus.cache.SideB has now a different result type; was: com.twitter.storehaus.cache.SideC#, is now: com.twitter.storehaus.cache.Side
- method side()com.twitter.storehaus.cache.SideC# in class com.twitter.storehaus.cache.SideCCyclicIncrement has now a different result type; was: com.twitter.storehaus.cache.SideC#, is now: com.twitter.storehaus.cache.Side
- method side()com.twitter.storehaus.cache.SideA# in class com.twitter.storehaus.cache.SideACyclicIncrement has now a different result type; was: com.twitter.storehaus.cache.SideA#, is now: com.twitter.storehaus.cache.Side
- method side()com.twitter.storehaus.cache.SideB# in class com.twitter.storehaus.cache.SideBCyclicIncrement has now a different result type; was: com.twitter.storehaus.cache.SideB#, is now: com.twitter.storehaus.cache.Side
- method nextSide()com.twitter.storehaus.cache.SideB# in object com.twitter.storehaus.cache.SideA has now a different result type; was: com.twitter.storehaus.cache.SideB#, is now: com.twitter.storehaus.cache.Side
- method nextSide()com.twitter.storehaus.cache.SideA# in object com.twitter.storehaus.cache.SideC has now a different result type; was: com.twitter.storehaus.cache.SideA#, is now: com.twitter.storehaus.cache.Side

Should we keep those as is or introduce mima exceptions?
- method multiMergeFromMultiSet(com.twitter.storehaus.Store,scala.collection.immutable.Map,com.twitter.storehaus.FutureCollector,com.twitter.algebird.Semigroup)scala.collection.immutable.Map in object com.twitter.storehaus.algebra.MergeableStore does not have a correspondent in new version

This wasn't introduced by this pr, should I go ahead and add an exception?

**Comment by johnynek** [comment]
looks like we made a mistake here: https://github.com/twitter/storehaus/releases/tag/v0.15.0-RC1 @pankajroark skipped over `0.14.0-RC2` which should have been the next version. I guess there can never be a `0.14.0` version now, and we should actually set `0.15.0-RC1` as the previous version.

About the `Side` issues, you can actually override with a more derived type, so you may be able to put the type: `SideA` etc... on those things to get mima clean.

That said, we can ignore those since we are going to be bumping the version anyway, we just want to minimize.

File: build.sbt [hunk 1]
```diff
@@ -112,7 +115,7 @@ val unreleasedModules = Set[String]()
 def youngestForwardCompatible(subProj: String) =
   Some(subProj)
     .filterNot(unreleasedModules.contains)
-    .map { s => "com.twitter" % ("storehaus-" + s + "_2.10") % "0.12.0" }
+    .map { s => "com.twitter" % ("storehaus-" + s + "_2.10") % "0.14.0-RC1" }
```
Comment (johnynek) [inline build.sbt]: I think we want to drop `_2.10` and let sbt do it: `"com.twitter" %% (s"storehaus-$s") % "0.15.0-RC1"`

Comment (BenFradet) [inline build.sbt]: will do

**Comment by BenFradet** [comment]
There should be no mima checks failing anymore except:

> (storehaus-memcache/*:mimaPreviousClassfiles) sbt.ResolveException: unresolved dependency: com.twitter.common.zookeeper#server-set;1.0.103: not found

which I haven't found out how to solve yet

File: build.sbt [hunk 2]
```diff
@@ -28,6 +29,29 @@ val testCleanup = Seq(
...
+      ".JavaFutureToTwitterFutureConverter$Open$"),
+    exclude[MissingClassProblem]("com.twitter.storehaus.kafka" +
+      ".JavaFutureToTwitterFutureConverter$State"),
+    exclude[MissingClassProblem]("com.twitter.storehaus.kafka" +
+      ".JavaFutureToTwitterFutureConverter$State$"),
+    exclude[MissingMethodProblem]("com.twitter.storehaus.kafka.KafkaStore.this"),
+    exclude[MissingMethodProblem]("com.twitter.storehaus.kafka.KafkaStore.apply")
+  )
+}
+
```
Comment (BenFradet) [inline build.sbt:54]: I excluded the future converter stuff, I don't know if this is the best way to go about things, input welcome.

**Comment by BenFradet** [comment]
This seems linked to [storehaus-memcache deps](https://github.com/twitter/storehaus/blob/develop/build.sbt#L182-L186), investigating.

**Comment by johnynek** [comment]
nice work! Much appreciated!

**Pull request merged**